.github/workflows/cicd.yml

```diff
@@ -0,0 +1,76 @@
+name: CICD
+
+on: [push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+
+jobs:
+
+  run-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout this repository
+        uses: actions/checkout@v2
+        with:
+            lfs: true
+
+      - name: Build application container
+        shell: bash -l {0}
+        run: docker build -t materials-designer:centos -f dockerfiles/centos/Dockerfile .
+
+      - name: Build test container
+        shell: bash -l {0}
+        run: docker build -t materials-designer:test -f dockerfiles/test/Dockerfile .
+
+      - name: Run tests
+        shell: bash -l {0}
+        run: |
+          docker run -p 3001:3001 -d materials-designer:centos
+          sleep 60
+          docker run --network=host materials-designer:test
+
+
+  publish:
+    needs: run-tests
+    runs-on: ubuntu-latest
+    if: github.ref_name == 'dev'
+
+    steps:
+      - name: Checkout this repository
+        uses: actions/checkout@v2
+        with:
+          lfs: true
+
+      - name: Build application container
+        shell: bash -l {0}
+        run: docker build -t materials-designer:centos -f dockerfiles/centos/Dockerfile .
+
+      - name: Build test container
+        shell: bash -l {0}
+        run: docker build -t materials-designer:test -f dockerfiles/test/Dockerfile .
+
+      - name: Run tests
+        shell: bash -l {0}
+        run: |
+          docker run -p 3001:3001 -d materials-designer:centos
+          sleep 30
+          docker run --network=host materials-designer:test
```

Comment:
Not a huge deal, but since we define them twice - might there be a way to reuse the equivalent actions we just defined above in this file?

Reply:
Definitely, I moved this set of logic into a local action in the workflows directory.